### PR TITLE
Adding `IsVerifiedDomain` to the `AzureADTenant` module

### PR DESCRIPTION
**Pull request by mynjj**
## What & why

Allows a consumer to validate whether a given domain is in the list of verified domains of a tenant, part of the [skip review for trusted senders slice](https://dynamicssmb2.visualstudio.com/Dynamics%20SMB/_workitems/edit/625413), as retrieved by the call to Graph's `/organization` endpoint

## Linked work

Fixes [AB#625413](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/625413)

## How I validated this

- [x] I read the full diff and it contains only changes I intended.
- [x]  I built the affected app(s) locally with no new analyzer warnings.
- [x] I ran the change in Business Central and confirmed it behaves as expected.

Usage works as expected:
image

## Risk & compatibility

Depends on [a change in LicensingService](https://microsoft.ghe.com/bic/BC-Dataplane-Services/pull/2693) and corresponding platform uptake. Without the corresponding LicensingService deployed, the list is empty (no crash), still it should only be *used* after full roll-out is validated.

### CODE REVIEW

**Comment by alexei-dobriansky** [comment]
## Agentic PR Review - Round 1

# **Recommendation: Request Changes**

<details>
<summary><strong>What this PR does</strong></summary>

This PR adds a public `Azure AD Tenant.GetVerifiedDomains()` procedure and an internal implementation that reads `TenantInfo.VerifiedDomains()` from the existing Azure AD Graph tenant detail. The change is additive: existing public signatures stay unchanged, and the new DotNet alias plus platform package bump provide the new model type. The main code path is narrow and filters null or empty domain names, so the implementation matches the slice goal, but the new public contract is not covered by an automated test.
</details>

<details>
<summary><strong>Suggestions</strong></summary>

**S1** - *Add coverage for verified domain API*
This PR adds a public System Application API that can affect trusted-sender decisions, but no automated test calls `GetVerifiedDomains()`. Please add a test with more than one verified domain and an empty-name case, so the public contract and filtering stay stable.
</details>

<details>
<summary><strong>Risk assessment and necessity</strong></summary>

**Risk:** The compatibility surface is broad because `Azure AD Tenant` is a public System Application codeunit. Verified domains are tenant metadata and may be privacy-sensitive, and consumers may use the result for security or trust decisions. The API is additive and not BaseApp-event dependent, but the returned list has no automated regression coverage.

**Necessity:** The linked work item is a Slice for trusted Payables Agent email scenarios, so exposing the tenant's verified domains is a coherent feature increment. The scope is small and the platform dependency is explicit through the package version update.
</details>

---
> [AI-PR-REVIEW] version=1 system=github pr=8499 round=1 by=alexei-dobriansky at=2026-07-02 lastSha=3e314d06da438aec9f75b2afcf9268308d264031 suggestions=S1

**Review by AndersLarsenMicrosoft — DISMISSED** [review]

File: src/System Application/App/Azure AD Tenant/src/AzureADTenantImpl.Codeunit.al [hunk 1]
```diff
@@ -62,6 +63,28 @@ codeunit 3705 "Azure AD Tenant Impl."
...
+        VerifiedDomain: DotNet VerifiedDomainInfo;
+        VerifiedDomains: DotNet GenericIEnumerable1;
+        VerifiedDomainName: Text;
+    begin
+        Initialize();
+        if IsNull(TenantInfo) then
+            Error(VerifiedDomainsErr);
+
+        VerifiedDomains := TenantInfo.VerifiedDomains();
+        if IsNull(VerifiedDomains) then
```
[the remark this answers is not in the document]
Comment (mynjj) [reply src/System Application/App/Azure AD Tenant/src/AzureADTenantImpl.Codeunit.al:77]: null here means that matching licensingservice is not deployed, while it shouldn't be the case this behavior prevents an unnecessary crash since this will be in any case a fallback

File: src/System Application/App/Azure AD Tenant/src/AzureADTenantImpl.Codeunit.al [hunk 2]
```diff
@@ -62,6 +63,28 @@ codeunit 3705 "Azure AD Tenant Impl."
         Error(PreferredLanguageErr);
     end;
 
+    procedure IsVerifiedDomain(Domain: Text): Boolean
```
[the remark this answers is not in the document]
Comment (mynjj) [reply src/System Application/App/Azure AD Tenant/src/AzureADTenantImpl.Codeunit.al:66]: there are no tests for this module, very small surface to be tested

**Comment by alexei-dobriansky** [comment]
## Agentic PR Review - Round 2

# **Recommendation: Request Changes**

<details>
<summary><strong>What this PR does</strong></summary>

Since round 1, the PR changed the public API from returning all verified domains to IsVerifiedDomain(Domain: Text): Boolean, added a null check for VerifiedDomains(), and scoped the new public method to OnPrem. The boolean API is a better security shape than returning the full domain list, and the comparison is case-insensitive. The change still has no automated coverage, and the new scope choice is not proven against the intended Payables Agent consumer.
</details>

<details>
<summary><strong>Status of previous suggestions</strong></summary>

| ID | Title | Status | Author response |
|----|-------|--------|-----------------|
| S1 | Add coverage for verified domain API | Disputed | A later comment says there are no tests for this module and the surface is small. I still consider coverage needed because this is a public API for a trust decision. |
</details>

<details>
<summary><strong>New observations (commits since round 1)</strong></summary>

**S2** - *Confirm the public API scope*
IsVerifiedDomain is the only public Azure AD Tenant method marked [Scope('OnPrem')], while the slice describes a Payables Agent scenario. Please either remove the scope or add a clear reason and proof that the intended caller can use this API.

**S3** - *Document the fallback result*
The XML doc says an error is shown when Graph cannot be reached, but the code returns alse when VerifiedDomains() is null. Please document this fallback so callers know that alse can also mean the verified-domain list was not available.
</details>

<details>
<summary><strong>Risk assessment and necessity</strong></summary>

**Risk:** This is a public System Application API with broad compatibility impact. It exposes tenant domain verification as a trust signal, so incorrect availability, fallback behavior, or missing regression coverage can affect sender-review decisions. The boolean shape limits data exposure, but [Scope('OnPrem')] may make the API unusable for the intended cloud-facing consumer.

**Necessity:** The linked work item is a Slice for Payables Agent trusted sender behavior, so a tenant-domain verification helper is a coherent feature increment. The scope is small, but the public contract must be stable and tested before merge.
</details>

---
> [AI-PR-REVIEW] version=1 system=github pr=8499 round=2 by=alexei-dobriansky at=2026-08-03T17:13:03Z lastSha=988aa5bfc1f5b0fb76f94827fcf445f7312b2832 suggestions=S1:disputed,S2:new,S3:new parentRound=1

**Comment by mynjj** [comment]
> ## Agentic PR Review - Round 2
> # **Recommendation: Request Changes**
> **What this PR does**
> Since round 1, the PR changed the public API from returning all verified domains to IsVerifiedDomain(Domain: Text): Boolean, added a null check for VerifiedDomains(), and scoped the new public method to OnPrem. The boolean API is a better security shape than returning the full domain list, and the comparison is case-insensitive. The change still has no automated coverage, and the new scope choice is not proven against the intended Payables Agent consumer.
> 
> **Status of previous suggestions**
> ID	Title	Status	Author response
> S1	Add coverage for verified domain API	Disputed	A later comment says there are no tests for this module and the surface is small. I still consider coverage needed because this is a public API for a trust decision.
> **New observations (commits since round 1)**
> **S2** - _Confirm the public API scope_ IsVerifiedDomain is the only public Azure AD Tenant method marked [Scope('OnPrem')], while the slice describes a Payables Agent scenario. Please either remove the scope or add a clear reason and proof that the intended caller can use this API.
> 
> **S3** - _Document the fallback result_ The XML doc says an error is shown when Graph cannot be reached, but the code returns alse when VerifiedDomains() is null. Please document this fallback so callers know that alse can also mean the verified-domain list was not available.
> 
> **Risk assessment and necessity**
> **Risk:** This is a public System Application API with broad compatibility impact. It exposes tenant domain verification as a trust signal, so incorrect availability, fallback behavior, or missing regression coverage can affect sender-review decisions. The boolean shape limits data exposure, but [Scope('OnPrem')] may make the API unusable for the intended cloud-facing consumer.
> 
> **Necessity:** The linked work item is a Slice for Payables Agent trusted sender behavior, so a tenant-domain verification helper is a coherent feature increment. The scope is small, but the public contract must be stable and tested before merge.
> 
> > [AI-PR-REVIEW] version=1 system=github pr=8499 round=2 by=alexei-dobriansky at=2026-08-03T17:13:03Z lastSha=988aa5bfc1f5b0fb76f94827fcf445f7312b2832 suggestions=S1:disputed,S2:new,S3:new parentRound=1

About S2, this is intentional, only needed in PA which is marked OnPrem, limiting the procedure to be on the safe side.
About S3, The fallback is also intentional in  the case the LicensingService is on a lower version (it shouldn't happen now, but still, a fail-safe)

**Review by WaelAbuSeada — APPROVED** [review]

**Review by ventselartur — APPROVED** [review]
